File: 2373-largest-local-values-in-a-matrix/2373-largest-local-values-in-a-matrix.py
class Solution:
    def largestLocal(self, grid: List[List[int]]) -> List[List[int]]:
        N = len(grid)
        res = [[0]*(N-2) for _ in range(N-2)]
        
        for i in range(N-2):
            for j in range(N-2):
                for r in range(i,i+3):
                    for c in range(j,j+3):
                        res[i][j] = max(res[i][j],grid[r][c])
        
        
        return res
        

File: 2373-largest-local-values-in-a-matrix/2373-largest-local-values-in-a-matrix.java
class Solution {
    public int[][] largestLocal(int[][] grid) {
        int n = grid.length;
        
        int[][] res= new int[n-2][n-2];
        for(int i=0;i<n-2;i++){
            for(int j=0;j<n-2;j++){
                for(int r=i;r<i+3;r++){
                    for(int c=j;c<j+3;c++){
                        res[i][j]=Math.max(res[i][j],grid[r][c]);
                    }
                }
            }
        }
        
        return res;
    }
}